File: lib/screens/menu_management_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../models/category_model.dart';
import '../models/menu_model.dart';
import '../providers/restaurant_provider.dart';
import '../repositories/restaurant_repository.dart';
import '../theme/app_colors.dart';
import '../widgets/category_filter_tabs.dart';

class MenuManagementScreen extends StatefulWidget {
  const MenuManagementScreen({super.key});

  @override
  State<MenuManagementScreen> createState() => _MenuManagementScreenState();
}

class _MenuManagementScreenState extends State<MenuManagementScreen> {
  int? _selectedCategoryId;

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<RestaurantProvider>();
    final categories = provider.categories;
    final menus = provider.menus;

    final filteredMenus = _selectedCategoryId == null
        ? menus
        : menus.where((m) => m.categoryId == _selectedCategoryId).toList();

    return Scaffold(
      backgroundColor: AppColors.surfaceLight,
      floatingActionButton: FloatingActionButton(
        onPressed: () => _showMenuFormDialog(context, provider, null),
        backgroundColor: AppColors.primary,
        foregroundColor: Colors.white,
        child: const Icon(Icons.add),
      ),
      body: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Header
          Container(
            color: Colors.white,
            width: double.infinity,
            padding: const EdgeInsets.fromLTRB(16, 16, 16, 12),
            child: const Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  'Kelola Menu & Stok Bahan',
                  style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold, color: AppColors.textPrimaryLight),
                ),
                Text(
                  'Atur harga jual, kategori, dan stok hidangan resto',
                  style: TextStyle(fontSize: 12, color: AppColors.textSecondaryLight),
                ),
              ],
            ),
          ),

          const Divider(height: 1, color: AppColors.surfaceLightBorder),

          // Categories Filter
          CategoryFilterTabs(
            categories: categories,
            selectedCategoryId: _selectedCategoryId,
            onCategorySelected: (catId) => setState(() => _selectedCategoryId = catId),
          ),

          // Menus List
          Expanded(
            child: ListView.builder(
              padding: const EdgeInsets.fromLTRB(16, 8, 16, 80),
              itemCount: filteredMenus.length,
              itemBuilder: (context, index) {
                final menu = filteredMenus[index];
                return _MenuItemCard(
                  menu: menu,
                  onStockChange: (newStock) => provider.updateMenuStock(menu.id!, newStock),
                  onEdit: () => _showMenuFormDialog(context, provider, menu),
                  onDelete: () => _confirmDeleteMenu(context, provider, menu),
                );
              },
            ),
          ),
        ],
      ),
    );
  }

  void _showMenuFormDialog(BuildContext context, RestaurantProvider provider, MenuModel? menu) {
    final nameCtrl = TextEditingController(text: menu?.name ?? '');
    final priceCtrl = TextEditingController(text: menu != null ? menu.price.toInt().toString() : '');
    final costPriceCtrl = TextEditingController(text: menu != null ? menu.costPrice.toInt().toString() : '');
    final stockCtrl = TextEditingController(text: menu != null ? menu.stock.toString() : '50');
    final descCtrl = TextEditingController(text: menu?.description ?? '');

    CategoryModel? selectedCategory = menu != null
        ? provider.categories.cast<CategoryModel?>().firstWhere((c) => c?.id == menu.categoryId, orElse: () => null)
        : (provider.categories.isNotEmpty ? provider.categories.first : null);

    showDialog(
      context: context,
      builder: (ctx) => StatefulBuilder(
        builder: (context, setModalState) => AlertDialog(
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
          title: Text(menu != null ? 'Edit Menu Makanan' : 'Tambah Menu Baru', style: const TextStyle(fontWeight: FontWeight.bold)),
          content: SingleChildScrollView(
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: [
                TextField(
                  controller: nameCtrl,
                  decoration: const InputDecoration(labelText: 'Nama Hidangan'),
                ),
                const SizedBox(height: 10),

                // Category Dropdown
                DropdownButtonFormField<CategoryModel>(
                  value: selectedCategory,
                  decoration: const InputDecoration(labelText: 'Kategori'),
                  items: provider.categories.map((cat) {
                    return DropdownMenuItem(value: cat, child: Text(cat.name));
                  }).toList(),
                  onChanged: (cat) => setModalState(() => selectedCategory = cat),
                ),
                const SizedBox(height: 10),

                Row(
                  children: [
                    Expanded(
                      child: TextField(
                        controller: priceCtrl,
                        keyboardType: TextInputType.number,
                        decoration: const InputDecoration(labelText: 'Harga Jual (Rp)'),
                      ),
                    ),
                    const SizedBox(width: 8),
                    Expanded(
                      child: TextField(
                        controller: stockCtrl,
                        keyboardType: TextInputType.number,
                        decoration: const InputDecoration(labelText: 'Stok Awal'),
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 10),

                TextField(
                  controller: descCtrl,
                  maxLines: 2,
                  decoration: const InputDecoration(labelText: 'Deskripsi Menu (Opsional)'),
                ),
              ],
            ),
          ),
          actions: [
            OutlinedButton(
              onPressed: () => Navigator.pop(ctx),
              child: const Text('Batal'),
            ),
            ElevatedButton(
              onPressed: () {
                if (nameCtrl.text.trim().isNotEmpty && selectedCategory != null && priceCtrl.text.isNotEmpty) {
                  if (menu != null) {
                    provider.updateMenu(
                      menu.copyWith(
                        name: nameCtrl.text.trim(),
                        categoryId: selectedCategory!.id,
                        categoryName: selectedCategory!.name,
                        price: double.tryParse(priceCtrl.text) ?? 0.0,
                        costPrice: double.tryParse(costPriceCtrl.text) ?? 0.0,
                        description: descCtrl.text.trim(),
                        stock: int.tryParse(stockCtrl.text) ?? 0,
                      ),
                    );
                  } else {
                    provider.addNewMenu(
                      name: nameCtrl.text.trim(),
                      categoryId: selectedCategory!.id!,
                      categoryName: selectedCategory!.name,
                      price: double.tryParse(priceCtrl.text) ?? 0.0,
                      costPrice: double.tryParse(costPriceCtrl.text) ?? 0.0,
                      description: descCtrl.text.trim(),
                      stock: int.tryParse(stockCtrl.text) ?? 0,
                    );
                  }
                  Navigator.pop(ctx);
                }
              },
              style: ElevatedButton.styleFrom(
                backgroundColor: AppColors.primary,
                foregroundColor: Colors.white,
              ),
              child: const Text('Simpan'),
            ),
          ],
        ),
      ),
    );
  }

  void _confirmDeleteMenu(BuildContext context, RestaurantProvider provider, MenuModel menu) {
    showDialog(
      context: context,
      builder: (ctx) => AlertDialog(
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
        title: const Text('Hapus Menu?', style: TextStyle(fontWeight: FontWeight.bold)),
        content: Text('Apakah Anda yakin ingin menghapus "${menu.name}"?'),
        actions: [
          OutlinedButton(
            onPressed: () => Navigator.pop(ctx),
            child: const Text('Batal'),
          ),
          ElevatedButton(
            onPressed: () {
              Navigator.pop(ctx);
              provider.deleteMenu(menu.id!, menu.name);
            },
            style: ElevatedButton.styleFrom(
              backgroundColor: AppColors.error,
              foregroundColor: Colors.white,
            ),
            child: const Text('Hapus'),
          ),
        ],
      ),
    );
  }
}

class _MenuItemCard extends StatelessWidget {
  final MenuModel menu;
  final ValueChanged<int> onStockChange;
  final VoidCallback onEdit;
  final VoidCallback onDelete;

  const _MenuItemCard({
    required this.menu,
    required this.onStockChange,
    required this.onEdit,
    required this.onDelete,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.only(bottom: 10),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(16),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withValues(alpha: 0.03),
            blurRadius: 6,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      padding: const EdgeInsets.all(14),
      child: Column(
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      menu.name,
                      style: const TextStyle(fontSize: 15, fontWeight: FontWeight.bold),
                    ),
                    Text(
                      '${menu.categoryName} • ${RestaurantRepository.formatRupiah(menu.price)}',
                      style: const TextStyle(
                        fontSize: 12,
                        color: AppColors.primary,
                        fontWeight: FontWeight.w600,
                      ),
                    ),
                  ],
                ),
              ),
              Row(
                mainAxisSize: MainAxisSize.min,
                children: [
                  IconButton(
                    icon: const Icon(Icons.edit, color: AppColors.primary, size: 18),
                    onPressed: onEdit,
                    constraints: const BoxConstraints(),
                    padding: const EdgeInsets.all(6),
                  ),
                  const SizedBox(width: 4),
                  IconButton(
                    icon: const Icon(Icons.delete, color: AppColors.error, size: 18),
                    onPressed: onDelete,
                    constraints: const BoxConstraints(),
                    padding: const EdgeInsets.all(6),
                  ),
                ],
              ),
            ],
          ),

          const SizedBox(height: 10),

          // Stock Stepper Bar
          Container(
            padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
            decoration: BoxDecoration(
              color: AppColors.surfaceLightElevated,
              borderRadius: BorderRadius.circular(10),
            ),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Row(
                  children: [
                    Icon(
                      Icons.inventory,
                      size: 16,
                      color: menu.stock > 0 ? AppColors.success : AppColors.error,
                    ),
                    const SizedBox(width: 6),
                    Text(
                      menu.stock > 0 ? 'Sisa Stok: ${menu.stock} ${menu.unit}' : 'Stok Habis (0)',
                      style: TextStyle(
                        fontSize: 12,
                        fontWeight: FontWeight.bold,
                        color: menu.stock > 0 ? AppColors.textPrimaryLight : AppColors.error,
                      ),
                    ),
                  ],
                ),

                Row(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    InkWell(
                      onTap: menu.stock > 0 ? () => onStockChange(menu.stock - 1) : null,
                      borderRadius: BorderRadius.circular(14),
                      child: Container(
                        width: 28,
                        height: 28,
                        decoration: const BoxDecoration(
                          color: AppColors.surfaceLightChip,
                          shape: BoxShape.circle,
                        ),
                        child: const Icon(Icons.remove, size: 14),
                      ),
                    ),
                    const SizedBox(width: 6),
                    InkWell(
                      onTap: () => onStockChange(menu.stock + 5),
                      borderRadius: BorderRadius.circular(14),
                      child: Container(
                        width: 28,
                        height: 28,
                        decoration: const BoxDecoration(
                          color: AppColors.primary,
                          shape: BoxShape.circle,
                        ),
                        child: const Icon(Icons.add, size: 14, color: Colors.white),
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

File: app/src/main/java/com/example/ui/screens/MenuManagementScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Inventory
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ExposedDropdownMenuBox
import androidx.compose.material3.ExposedDropdownMenuDefaults
import androidx.compose.material3.FilledTonalIconButton
import androidx.compose.material3.FloatingActionButton
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.IconButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.CategoryEntity
import com.example.data.model.MenuEntity
import com.example.data.repository.RestaurantRepository
import com.example.ui.components.CategoryFilterTabs
import com.example.ui.theme.AmberPrimary
import com.example.ui.theme.ErrorRed
import com.example.ui.theme.SuccessGreen
import com.example.ui.viewmodel.RestaurantViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun MenuManagementScreen(
    categories: List<CategoryEntity>,
    menus: List<MenuEntity>,
    viewModel: RestaurantViewModel,
    modifier: Modifier = Modifier
) {
    var selectedCategoryId by remember { mutableStateOf<Int?>(null) }
    var showAddMenuDialog by remember { mutableStateOf(false) }
    var menuToEdit by remember { mutableStateOf<MenuEntity?>(null) }

    val filteredMenus = if (selectedCategoryId == null) menus else menus.filter { it.categoryId == selectedCategoryId }

    Scaffold(
        modifier = modifier.fillMaxSize(),
        floatingActionButton = {
            FloatingActionButton(
                onClick = { showAddMenuDialog = true },
                containerColor = AmberPrimary,
                contentColor = Color.White,
                shape = RoundedCornerShape(16.dp),
                modifier = Modifier.testTag("add_menu_fab")
            ) {
                Icon(imageVector = Icons.Default.Add, contentDescription = "Tambah Menu Baru")
            }
        }
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
        ) {
            // Header
            Surface(
                modifier = Modifier.fillMaxWidth(),
                color = MaterialTheme.colorScheme.surface,
                tonalElevation = 2.dp
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text(
                        text = "Kelola Menu & Stok Bahan",
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurface
                    )
                    Text(
                        text = "Atur harga jual, kategori, dan stok hidangan resto",
                        fontSize = 12.sp,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            // Categories Filter
            CategoryFilterTabs(
                categories = categories,
                selectedCategoryId = selectedCategoryId,
                onCategorySelected = { selectedCategoryId = it }
            )

            // Menus List
            LazyColumn(
                modifier = Modifier
                    .fillMaxWidth()
                    .weight(1f)
                    .padding(horizontal = 16.dp),
                contentPadding = PaddingValues(top = 8.dp, bottom = 80.dp),
                verticalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                items(filteredMenus, key = { it.id }) { menu ->
                    MenuManagementItemCard(
                        menu = menu,
                        onStockChange = { newStock -> viewModel.updateMenuStock(menu.id, newStock) },
                        onEdit = { menuToEdit = menu },
                        onDelete = { viewModel.deleteMenu(menu) }
                    )
                }
            }
        }
    }

    // Add or Edit Menu Dialog
    if (showAddMenuDialog || menuToEdit != null) {
        MenuFormDialog(
            menu = menuToEdit,
            categories = categories,
            onDismiss = {
                showAddMenuDialog = false
                menuToEdit = null
            },
            onSave = { name, catId, catName, price, costPrice, desc, stock ->
                if (menuToEdit != null) {
                    viewModel.updateMenu(
                        menuToEdit!!.copy(
                            name = name,
                            categoryId = catId,
                            categoryName = catName,
                            price = price,
                            costPrice = costPrice,
                            description = desc,
                            stock = stock
                        )
                    )
                } else {
                    viewModel.addNewMenu(
                        name = name,
                        categoryId = catId,
                        categoryName = catName,
                        price = price,
                        costPrice = costPrice,
                        description = desc,
                        stock = stock
                    )
                }
                showAddMenuDialog = false
                menuToEdit = null
            }
        )
    }
}

@Composable
private fun MenuManagementItemCard(
    menu: MenuEntity,
    onStockChange: (Int) -> Unit,
    onEdit: () -> Unit,
    onDelete: () -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(14.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.5.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(14.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = menu.name,
                        fontSize = 15.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurface
                    )
                    Text(
                        text = "${menu.categoryName} • ${RestaurantRepository.formatRupiah(menu.price)}",
                        fontSize = 12.sp,
                        color = AmberPrimary,
                        fontWeight = FontWeight.SemiBold
                    )
                }

                Row {
                    IconButton(onClick = onEdit, modifier = Modifier.size(32.dp)) {
                        Icon(Icons.Default.Edit, contentDescription = "Edit", tint = AmberPrimary, modifier = Modifier.size(18.dp))
                    }
                    IconButton(onClick = onDelete, modifier = Modifier.size(32.dp)) {
                        Icon(Icons.Default.Delete, contentDescription = "Hapus", tint = ErrorRed, modifier = Modifier.size(18.dp))
                    }
                }
            }

            Spacer(modifier = Modifier.height(10.dp))

            // Stock Quick Editor Bar
            Surface(
                shape = RoundedCornerShape(10.dp),
                color = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.4f),
                modifier = Modifier.fillMaxWidth()
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 12.dp, vertical = 6.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Default.Inventory,
                            contentDescription = null,
                            tint = if (menu.stock > 0) SuccessGreen else ErrorRed,
                            modifier = Modifier.size(16.dp)
                        )
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(
                            text = if (menu.stock > 0) "Sisa Stok: ${menu.stock} ${menu.unit}" else "Stok Habis (0)",
                            fontSize = 12.sp,
                            fontWeight = FontWeight.Bold,
                            color = if (menu.stock > 0) MaterialTheme.colorScheme.onSurface else ErrorRed
                        )
                    }

                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(4.dp)
                    ) {
                        FilledTonalIconButton(
                            onClick = { if (menu.stock > 0) onStockChange(menu.stock - 1) },
                            modifier = Modifier.size(28.dp),
                            shape = CircleShape
                        ) {
                            Icon(Icons.Default.Remove, contentDescription = "Kurang Stok", modifier = Modifier.size(14.dp))
                        }

                        FilledTonalIconButton(
                            onClick = { onStockChange(menu.stock + 5) },
                            modifier = Modifier.size(28.dp),
                            shape = CircleShape,
                            colors = IconButtonDefaults.filledTonalIconButtonColors(containerColor = AmberPrimary, contentColor = Color.White)
                        ) {
                            Icon(Icons.Default.Add, contentDescription = "Tambah Stok", modifier = Modifier.size(14.dp))
                        }
                    }
                }
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun MenuFormDialog(
    menu: MenuEntity?,
    categories: List<CategoryEntity>,
    onDismiss: () -> Unit,
    onSave: (String, Int, String, Double, Double, String, Int) -> Unit
) {
    var name by remember { mutableStateOf(menu?.name ?: "") }
    var selectedCategory by remember { mutableStateOf(categories.find { it.id == menu?.categoryId } ?: categories.firstOrNull()) }
    var priceText by remember { mutableStateOf(menu?.price?.toLong()?.toString() ?: "") }
    var costPriceText by remember { mutableStateOf(menu?.costPrice?.toLong()?.toString() ?: "") }
    var stockText by remember { mutableStateOf(menu?.stock?.toString() ?: "50") }
    var description by remember { mutableStateOf(menu?.description ?: "") }
    var isCategoryDropdownExpanded by remember { mutableStateOf(false) }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = {
            Text(
                text = if (menu != null) "Edit Menu Makanan" else "Tambah Menu Baru",
                fontWeight = FontWeight.Bold
            )
        },
        text = {
            Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
                OutlinedTextField(
                    value = name,
                    onValueChange = { name = it },
                    label = { Text("Nama Hidangan") },
                    singleLine = true,
                    shape = RoundedCornerShape(12.dp),
                    modifier = Modifier.fillMaxWidth()
                )

                // Category selector
                ExposedDropdownMenuBox(
                    expanded = isCategoryDropdownExpanded,
                    onExpandedChange = { isCategoryDropdownExpanded = it }
                ) {
                    OutlinedTextField(
                        value = selectedCategory?.name ?: "Pilih Kategori",
                        onValueChange = {},
                        readOnly = true,
                        label = { Text("Kategori") },
                        trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded = isCategoryDropdownExpanded) },
                        shape = RoundedCornerShape(12.dp),
                        modifier = Modifier
                            .fillMaxWidth()
                            .menuAnchor()
                    )
                    ExposedDropdownMenu(
                        expanded = isCategoryDropdownExpanded,
                        onDismissRequest = { isCategoryDropdownExpanded = false }
                    ) {
                        categories.forEach { category ->
                            DropdownMenuItem(
                                text = { Text(category.name) },
                                onClick = {
                                    selectedCategory = category
                                    isCategoryDropdownExpanded = false
                                }
                            )
                        }
                    }
                }

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    OutlinedTextField(
                        value = priceText,
                        onValueChange = { priceText = it.filter { ch -> ch.isDigit() } },
                        label = { Text("Harga Jual (Rp)") },
                        singleLine = true,
                        shape = RoundedCornerShape(12.dp),
                        modifier = Modifier.weight(1f)
                    )
                    OutlinedTextField(
                        value = stockText,
                        onValueChange = { stockText = it.filter { ch -> ch.isDigit() } },
                        label = { Text("Stok Awal") },
                        singleLine = true,
                        shape = RoundedCornerShape(12.dp),
                        modifier = Modifier.weight(1f)
                    )
                }

                OutlinedTextField(
                    value = description,
                    onValueChange = { description = it },
                    label = { Text("Deskripsi Menu (Opsional)") },
                    shape = RoundedCornerShape(12.dp),
                    modifier = Modifier.fillMaxWidth()
                )
            }
        },
        confirmButton = {
            Button(
                onClick = {
                    if (name.isNotBlank() && selectedCategory != null && priceText.isNotBlank()) {
                        onSave(
                            name.trim(),
                            selectedCategory!!.id,
                            selectedCategory!!.name,
                            priceText.toDoubleOrNull() ?: 0.0,
                            costPriceText.toDoubleOrNull() ?: 0.0,
                            description.trim(),
                            stockText.toIntOrNull() ?: 0
                        )
                    }
                },
                colors = ButtonDefaults.buttonColors(containerColor = AmberPrimary),
                shape = RoundedCornerShape(10.dp)
            ) {
                Text("Simpan")
            }
        },
        dismissButton = {
            OutlinedButton(onClick = onDismiss, shape = RoundedCornerShape(10.dp)) {
                Text("Batal")
            }
        },
        shape = RoundedCornerShape(18.dp)
    )
}
